### DISCUSSION TITLE
How to run private images from Docker Hub?

### DISCUSSION THREAD

**Discussion by bounlu** [nextflow-io/nextflow#2939] [Q&A] [answered]
As the title says, how can I run private images from Docker Hub inside Nextflow pipeline to fix the below error?

Where to provide credentials or `docker login` command? Does `containerOptions` support this?

```
process {
	withName: 'XXX' {
		container	= 'personal/privateimage:latest'
	}
}

Caused by:
  Task failed to start - CannotPullContainerError: Error response from daemon: pull access denied for personal/privateimage:latest, repository does not exist or may require 'docker login': denied: requested access to the resource is denied
```

**Comment by abhi18av** [comment]
Hi @bounlu 

Could you please share more context with us, such as `.nextflow.log` file would be really helpful.


Beyond this, are you trying to run this locally or in cloud?

**Comment by bounlu** [comment]
Hi @abhi18av 

Unfortunately I don't have the `.nextflow.log` file anymore but I believe this is a general issue.

I am running it on the cloud via `awsbatch`.

**Comment by bounlu** [comment, marked as answer]
Any suggestions on this?

**Reply by pditommaso** [reply]
Nextflow currently does not provide direct support for private container repositories. However, the use of private containers depending the capability of the target compute platform. 

For example this post describes how to authenticate docker when using AWS batch https://aws.amazon.com/blogs/compute/how-to-authenticate-private-container-registries-using-aws-batch/

**Reply by bounlu** [reply]
@pditommaso 

Great! Following your link I could manage to set it up. Basically the idea is to place DockerHub credentials inside the AMI on AWS which will be used for authentication when pulling private images, as explained [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/private-auth-container-instances.html).

Thanks a lot.
